### PR TITLE
Support nested `@body`

### LINKED ISSUE #1141: Should nested @body be allowed
Currently this fails 
```cadl
import "@cadl-lang/rest";

using Cadl.Http;

model Image {
  @header contentType: "application/octet-stream";
  @body body: bytes;
}

op doStuffWithBytes(data: Image): int32;
```
with 
```
Operation has a @body and an unannotated parameter. There can only be one representing the body
```

This error is a bit confusing, believe we should either disallow nested @body and give a relevant error or this should be working.

[Playground Link](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwoKdXNpbmcgQ2FkbC5IdHRwOwoKbW9kZWwgSW1hZ2UgewogIEBoZWFkZXIoImNvbnRlbnQtdHlwZSIpIMcPVHlwZTogImFwcGxpY2F0aW9uL29jdGV0LXN0cmVhbSI7xENib2R5IMQFOiBieXRlczsKfQoKb3AgZG9TdHVmZldpdGhCxBgoZGF0YTrGfSk6IGludDMyOwo%3D)

### PR DESCRIPTION
**Pull request by nguerrera**
Fix #1141 

This was originally intended to be supported and got broken during development when we added automatic body for unannotated properties at the same time. This is why the error message is hard to understand. This makes it work. The issue is still tagged design, but I consider it a bug fix.

### CODE REVIEW

File: packages/http/src/metadata.ts
Comment (timotheeguerin) [inline packages/http/src/metadata.ts:1]: I see it ignores other properties in the nested model, is it expected
https://cadlplayground.z22.web.core.windows.net/prs/2029/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKLy8gQHJvdXRlKCJpbXBsaWNpdCIpIG9wIMgOKGE6IHN0cmluZywgYsgLKTogdm9pZM5AZmFpbC0xxj7EDDHMO0Bib2R530HFQTLKQTIoxB4gIMpHxhBiOiB7xgogx1hjyCI7yBhkzxJ9xAfKds1zM8pzMygKx2DVXc9axFfJVA%3D%3D

in the playground you can uncomment the other fail operation and those fail as expected but not if you have 

```
 op fail3(
  b: {
    @body c: string;
    d: string;
  }
): void;
```

Comment (nguerrera) [reply packages/http/src/metadata.ts:1]: Hmm, FWIW, I think `@body` has had the impact already of dropping other unannotated properties on the response.

https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKQHJvdXRlKCJ0ZXN0IikKb3AgxAooKTogewogIGE6IHN0cmluZzsKICBAYm9keSBiyhN9Ow%3D%3D

We only consider top-level parameters unannotated. We could perhaps make this an error, but it feels like then we should do the same on response.

Comment (timotheeguerin) [reply packages/http/src/metadata.ts:1]: hhm yeah feels like the response here should be consistent

Comment (timotheeguerin) [reply packages/http/src/metadata.ts:1]: I think this should be an error (or at least a warning) some of the data is being lost here

Comment (nguerrera) [reply packages/http/src/metadata.ts:1]: Trying to figure out how far back that response behavior goes, I think to the inception of `@body`

Comment (timotheeguerin) [reply packages/http/src/metadata.ts:1]: This already being inconsistent maybe we don't need to do that in this PR though and can file an issue about it.

**Review by timotheeguerin — APPROVED** [review]
This looked good I think just need a new issue to talk about what should we do if there is extra unannoted properties and `@body` for the nested case and in response

File: packages/http/src/metadata.ts
Comment (nguerrera) [reply packages/http/src/metadata.ts:1]: https://github.com/microsoft/typespec/issues/2075

**Pull request merged**